src/lighteval/models/base_model.py

```diff
@@ -541,10 +545,14 @@ def greedy_until(
             for batch in tqdm(
                 dataloader, desc="Greedy generation", position=1, leave=False, disable=self.disable_tqdm
             ):
-                # NOTE: we are assuming all items in a batch behave similarly (same
-                # stop_tokens and max_tokens genrated) which is not necessarily
-                # the case! Because of that we only use batch size of 1
-                stop_tokens = batch[0].stop_sequence
+                # For chat models, generation stops with EOS token, so we don't need to specify stop tokens
+                if hasattr(self.tokenizer, "chat_template") and self.tokenizer.chat_template is not None:
```

Comment:
Ideally we would check if `args.use_chat_template` is true here - is there a way to proagate that?